pandas/_libs/tslibs/period.pyx

```diff
@@ -1195,6 +1196,10 @@ cdef class _Period(object):
             freq = self._maybe_convert_freq(freq)
         how = _validate_end_alias(how)
 
```

Comment:
do we have an assert on how? (e.g. it has to be S/E)? if not can you add one

Reply:
This is done in ``_validate_end_alias`` I think:

https://github.com/pandas-dev/pandas/blob/ad50b1d77531f41a9bd684a1eff8bfe2831c24b8/pandas/_libs/tslibs/period.pyx#L1843-L1850